### PR TITLE
PHP8 Support

### PR DESCRIPTION
**Pull request by flavioheleno**
This PR adds support for PHP8.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,11 +3,17 @@ dist: trusty
...
 php:
     - '7.4'
+    - nightly
 
 matrix:
     include:
         - php: '7.4'
           env: 'COMPOSER_FLAGS="--prefer-stable --prefer-lowest"'
+        - php: nightly
+          env: 'COMPOSER_FLAGS="--ignore-platform-reqs"'
```
Comment (GrahamCampbell) [inline .travis.yml:13]: We should avoid this.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (flavioheleno) [reply .travis.yml:13]: Hi @GrahamCampbell I agree, this is just while not all deps support PHP8, I'll remove it from the PR at a later point :-)

Comment (GrahamCampbell) [reply .travis.yml:13]: Which deps don't support PHP 8?

Comment (flavioheleno) [reply .travis.yml]: Just removed as I checked all deps already support PHP 8 :-D

**Review by rosstuck — APPROVED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -3,11 +3,17 @@ dist: trusty
...
 
 matrix:
     include:
         - php: '7.4'
           env: 'COMPOSER_FLAGS="--prefer-stable --prefer-lowest"'
+        - php: nightly
+          env: 'COMPOSER_FLAGS="--prefer-stable --prefer-lowest"'
+    fast_finish: true
+    allow_failures:
+        - php: nightly
```
Comment (rosstuck) [inline .travis.yml:16]: Nice, didn't know about these options but that's really handy :smile:

**Comment by rosstuck** [comment]
Excellent, thanks once again! :+1:

**Pull request merged**